src/main/java/com/oath/halodb/DBDirectory.java

```diff
@@ -29,14 +29,25 @@ private DBDirectory(File dbDirectory, FileChannel directoryChannel) {
 
     /**
      * Will create a new directory if one doesn't already exist.
+     * The channel is created to attempt fsync. Windows is known not to allow file channels on
```

Comment:
I think we don't need this comment here as this has already been explained in the same file [here](https://github.com/yahoo/HaloDB/blob/master/src/main/java/com/oath/halodb/DBDirectory.java#L59).

You could move that comment in `syncMetaDataa` to `openReadOnlyChannel`